File: chatuikit/src/main/java/com/cometchat/chatuikit/shared/spans/BoldFormatSpan.java
package com.cometchat.chatuikit.shared.spans;

import android.graphics.Typeface;
import android.os.Build;
import android.text.TextPaint;
import android.text.style.MetricAffectingSpan;

import androidx.annotation.NonNull;

import com.cometchat.chatuikit.shared.views.richtexttoolbar.FormatType;

/**
 * Span for bold formatting in WYSIWYG rich text editing.
 * <p>
 * This span uses font weight 700 (API 28+) or {@link Typeface#BOLD} fallback
 * to render text in bold style. It implements {@link RichTextFormatSpan} to
 * enable format detection and markdown conversion.
 * </p>
 *
 * @see RichTextFormatSpan
 * @see FormatType#BOLD
 */
public class BoldFormatSpan extends MetricAffectingSpan implements RichTextFormatSpan {

    private static final int BOLD_WEIGHT = 700;

    /**
     * Creates a new BoldFormatSpan that renders text with font weight 700.
     */
    public BoldFormatSpan() {
    }

    @Override
    public void updateMeasureState(@NonNull TextPaint textPaint) {
        applyBold(textPaint);
    }

    @Override
    public void updateDrawState(@NonNull TextPaint textPaint) {
        applyBold(textPaint);
    }

    private void applyBold(@NonNull TextPaint paint) {
        Typeface currentTypeface = paint.getTypeface();
        boolean isItalic = currentTypeface != null && currentTypeface.isItalic();

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
            Typeface base = currentTypeface != null ? currentTypeface : Typeface.DEFAULT;
            paint.setTypeface(Typeface.create(base, BOLD_WEIGHT, isItalic));
        } else {
            int style = isItalic ? Typeface.BOLD_ITALIC : Typeface.BOLD;
            Typeface base = currentTypeface != null ? currentTypeface : Typeface.DEFAULT;
            paint.setTypeface(Typeface.create(base, style));
        }
    }

    /**
     * Returns the format type associated with this span.
     *
     * @return {@link FormatType#BOLD} indicating this is a bold format span.
     */
    @Override
    public FormatType getFormatType() {
        return FormatType.BOLD;
    }
}

File: chatuikit-kotlin/src/main/java/com/cometchat/uikit/kotlin/shared/spans/BoldFormatSpan.kt
package com.cometchat.uikit.kotlin.shared.spans

import android.graphics.Typeface
import android.os.Build
import android.text.TextPaint
import android.text.style.MetricAffectingSpan
import com.cometchat.uikit.core.formatter.RichTextFormat

/**
 * Span for bold formatting in WYSIWYG rich text editing.
 *
 * Uses font weight 700 (API 28+) or [Typeface.BOLD] fallback to render text
 * in bold style. Implements [RichTextFormatSpan] to enable format detection
 * and markdown conversion.
 *
 * @see RichTextFormatSpan
 * @see RichTextFormat.BOLD
 */
class BoldFormatSpan : MetricAffectingSpan(), RichTextFormatSpan {

    override fun updateMeasureState(textPaint: TextPaint) {
        applyBold(textPaint)
    }

    override fun updateDrawState(textPaint: TextPaint) {
        applyBold(textPaint)
    }

    private fun applyBold(paint: TextPaint) {
        val currentTypeface = paint.typeface
        val isItalic = currentTypeface?.isItalic == true

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
            val base = currentTypeface ?: Typeface.DEFAULT
            paint.typeface = Typeface.create(base, BOLD_WEIGHT, isItalic)
        } else {
            val style = if (isItalic) Typeface.BOLD_ITALIC else Typeface.BOLD
            val base = currentTypeface ?: Typeface.DEFAULT
            paint.typeface = Typeface.create(base, style)
        }
    }

    override fun getFormatType(): RichTextFormat = RichTextFormat.BOLD

    companion object {
        private const val BOLD_WEIGHT = 700
    }
}
